### PR TITLE
feature: network switcher

### PR DESCRIPTION
**Pull request by delaaxe**
Adds a network switcher to the account page and uses the corresponding network in wallet & page contents

### CODE REVIEW

**Comment by juniset** [comment]
When I switch to mainnet I land on the Account screen with the possibility to create a new account. There is no way to cancel the action and go back (I'm forced to create an account). We should add a back arrow in the top left corner or a cancel button in the bottom.

**Comment by delaaxe** [comment]
> When I switch to mainnet I land on the Account screen with the possibility to create a new account. There is no way to cancel the action and go back (I'm forced to create an account). We should add a back arrow in the top left corner or a cancel button in the bottom.

But if you click on those, to which wallet should it bring you back? My thought was adding the network switcher to the account list page too

File: packages/extension/package.json [hunk 1]
```diff
@@ -41,7 +41,7 @@
     "react-hook-form": "^7.19.4",
     "react-promise-suspense": "^0.3.3",
     "react-router-dom": "^6.0.1",
-    "starknet": "^2.0.1",
+    "starknet": "seanjameshan/starknet.js#develop",
```
Comment (janek26) [inline packages/extension/package.json:44]: should this go to main branch? 
No other way for now, as starknet.js deployment is still blocked, but I really don't like this :/ can we maybe create an issue to replace this ASAP ?

File: packages/extension/src/background/keys/l1.ts [hunk 2]
```diff
@@ -137,10 +120,25 @@ function downloadTextFile(text: string, filename: string) {
   })
 }
 
-export const getWallets = () => store.getItem("walletAddresses")
+export const getWallets = async (networkId?: string): Promise<BackupWallet[]> =>
+  (await store.getItem("walletAddresses"))
+    .map((wallet) => {
+      if (typeof wallet === "string") {
+        // backwards compatibility
```
Comment (janek26) [inline packages/extension/src/background/keys/l1.ts:127]:
```suggestion
        // TODO: remove backwards compatibility
```

File: packages/extension/src/inpage/index.ts [hunk 3]
```diff
@@ -70,22 +72,37 @@ const starknetWindowObject: StarknetWindowObject = {
...
 window.starknet = starknetWindowObject
 
+window.addEventListener(
+  "message",
+  ({ data }: MessageEvent<WindowMessageType>) => {
+    const { starknet } = window
+    if (starknet && starknet.signer && data.type === "CHANGE_NETWORK") {
+      console.warn("CHANGE_NETWORK", data.data)
+      starknet.provider = new Provider({ network: data.data as any })
+      starknet.signer = new WalletSigner(
```
Comment (janek26) [inline packages/extension/src/inpage/index.ts:82]: should we also set `starknet.selectedAddress` (as it probably has changed) here?

File: packages/extension/package.json (hunk 1 above)
Comment (delaaxe) [reply packages/extension/package.json:44]: OK let's wait then maybe

File: packages/extension/src/inpage/index.ts (hunk 3 above)
Comment (delaaxe) [reply packages/extension/src/inpage/index.ts:82]: also this should only be broadcasted to the page if there is an actual address...

**Review by janek26 — APPROVED** [review]

**Comment by janek26** [comment]
great work @delaaxe !

**Pull request merged**